Libraries/LibWeb/HTML/ValidityState.cpp

```diff
@@ -6,14 +6,21 @@
 
 #include <LibWeb/Bindings/Intrinsics.h>
 #include <LibWeb/Bindings/ValidityStatePrototype.h>
+#include <LibWeb/HTML/HTMLElement.h>
 #include <LibWeb/HTML/ValidityState.h>
 
 namespace Web::HTML {
 
 GC_DEFINE_ALLOCATOR(ValidityState);
 
-ValidityState::ValidityState(JS::Realm& realm)
+GC::Ref<ValidityState> ValidityState::create(JS::Realm& realm, FormAssociatedElement const* control)
+{
+    return realm.create<ValidityState>(realm, control);
+}
+
+ValidityState::ValidityState(JS::Realm& realm, FormAssociatedElement const* control)
     : PlatformObject(realm)
+    , m_control(control)
```

Comment:
Could this be passed in and stored as a reference instead of a pointer, since it is never null?

Reply:
Yes it could, done.